src/views/StoryViewer.vue

```diff
@@ -19,31 +19,76 @@ export default {
   name: 'StoryViewer',
   data: () => {
     return {
-      frameDuration: 2000,
-      storyEnd: false
+      frameDuration: 6000,
+      frameStarting: null,
+      frameRemaining: null,
+      storyEnd: false,
+      currentTimeout: null,
+      isPaused: false
     }
   },
   components: {
     ImageAttribution
   },
-  computed: mapGetters(['currentFrame', 'storyLength']),
+  computed: {
+    ...mapGetters(['currentFrame', 'storyLength']),
+    animationDuration: function () {
+      return this.frameDuration / 1000 + 's'
+    }
+  },
   methods: {
     ...mapActions(['selectFrame']),
+    setFrameTimeout: function(f) {
+      const duration = this.frameRemaining ? this.frameRemaining : this.frameDuration
+      this.frameStarting = Date.now()
+      this.currentTimeout = setTimeout( () => {
+        f()
+        clearTimeout(this.currentTimeout)
+        this.frameRemaining = null
+        this.frameStarting = null
+        this.currentTimeout = null
+      }, duration)
+    },
     playNextFrame: function() {
-      const timeoutId = setTimeout( () => {
-        this.selectFrame(this.currentFrame.id + 1)
-        clearTimeout(timeoutId)
-      }, this.frameDuration)
+      const playNext = () => this.selectFrame(this.currentFrame.id + 1)
+      this.setFrameTimeout(playNext)
     },
     restartStory: function() {
       this.storyEnd = false
       this.selectFrame(1)
+      this.frameRemaining = null
+      this.frameStarting = null
+      this.currentTimeout = null
     },
     endStory: function() {
-      const timeoutId = setTimeout( ()=> {
-        this.storyEnd = true
-        clearTimeout(timeoutId)
-      }, this.frameDuration)
+      const end = () => this.storyEnd = true
+      this.setFrameTimeout(end)
+    },
+    isPauseAction: function(e) {
+      const invalidClick = e.target.className === 'restart-btn'
+      return !invalidClick && !this.storyEnd
+    },
+    beginPause: function(e) {
+      this.frameRemaining = this.frameDuration - Math.abs(this.frameStarting - Date.now())
+      if (this.isPauseAction(e)) {
+        e.preventDefault()
+        e.stopPropagation()
+        clearTimeout(this.currentTimeout)
+        this.currentTimeout = null
+        this.isPaused = true
+      }
+    },
+    endPause: function(e) {
+      if (this.isPauseAction(e)) {
+        e.preventDefault()
+        e.stopPropagation()
+        this.isPaused = false
+      }
+    },
+    handlePause: function(e) {
```

Comment:
From the logic of the function, I would say "toggle" represent what it does more than "handle".